internal/engine/upper.go

```diff
@@ -484,13 +487,18 @@ func handleConfigsReloaded(
 		}
 
 		newDefOrder[i] = m.ManifestName()
+
+		// reset which files have caused a change
+		mt.State.ConfigFilesThatCausedChange = []string{}
+		configFilesThatChanged := state.LastTiltfileBuild.Edits
 		if !m.Equal(mt.Manifest) {
 			mt.Manifest = m
 
 			// Manifest has changed, ensure we do an image build so that we apply the changes
 			state := mt.State
 			state.BuildStatuses = make(map[model.TargetID]*store.BuildStatus)
 			state.PendingManifestChange = time.Now()
+			state.ConfigFilesThatCausedChange = configFilesThatChanged
```

Comment:
it might be better to update state.PendingFiles? And then change BuildController so that it knows this is not really the first build if there are existing PendingFiles?

Part of the problem, AFAICT, is that BuildController thinks this is the first build when the manifest gets cleared.